plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3462,6 +3423,22 @@ public void testDeleteOrphanFilesParameterValidation()
                 "\\QRetention specified (33.00s) is shorter than the minimum retention configured in the system (7.00d). Minimum retention can be changed with iceberg.delete_orphan_files.min-retention configuration property or iceberg.delete_orphan_files_min_retention session property");
     }
 
+    @Test
+    public void testIfDeletesReturnsNumberOfRemovedRows()
+    {
+        String tableName = "test_delete_returns_number_of_rows_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (key varchar, value integer) WITH (partitioning = ARRAY['key'])");
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('one', 1)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('one', 2)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('one', 3)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('two', 1)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('two', 2)", 1);
+        assertUpdate("DELETE FROM " + tableName + " WHERE key = 'one'", 3);
+        assertUpdate("DELETE FROM " + tableName + " WHERE key = 'one'");
+        assertUpdate("DELETE FROM " + tableName + " WHERE key = 'three'");
```

Comment:
can we make Iceberg fill DELETED_RECORDS_PROP always?
let's file a PR and link with a TODO note here.

Reply:
thanks for adding a TODO note. Can it contain a link?

Reply:
I don't have a pr yet. Should I open an issue? What if they decide to close it ?

Reply:
let's have an issue.

even if it gets closed, it will serve as an anchor for further information and current status